src/simulators/density_matrix/densitymatrix_state.hpp

```diff
@@ -297,6 +299,18 @@ class State : public Base::State<densmat_t> {
 
   // Table of allowed snapshot types to enum class members
   const static stringmap_t<Snapshots> snapshotset_;
+
+  //scale for density matrix = 2
+  virtual int qubit_scale(void)
```

Comment:
Is this actually used anywhere?

Reply:
Currently multi-shots optimization is only supported if device=GPU. And Thrust(CPU) is not supported. So we can not test batched_shots_optimization on CI. 
batched_shots_optimization is enabled by default for smaller number of qubits, so when running tests on local machine with GPU, all the test cases that require multiple shots use this option. (all the test is passed on local machine.)